keras/models.py

```diff
@@ -1432,7 +1438,15 @@ def input_validation(generator_output):
                 raise Exception('All input arrays and target arrays must have '
                                 'the same number of samples.')
             return data, sample_weight
-
+        if do_validation:
+            data_val, sample_weight_val = input_validation(validation_data)
+            sample_weight_val_l = [standardize_weights(data_val[name],
```

Comment:
For the `Graph`, I need the `sample_weight` to be a `dict` and I create a `list` so it has the right format for `self.validation_data`. I keep the `dict` so it could be passed to evaluate. Should I move the normalization of weights to `input_validation` and return a `list` and a `dict`?